python_modules/dagster/dagster/core/asset_defs/assets.py

```diff
@@ -191,6 +198,47 @@ def with_replaced_asset_keys(
                 partition_mappings=self._partition_mappings,
             )
 
+    def to_source_assets(self) -> Sequence[SourceAsset]:
+        result = []
+        for output_name, asset_key in self.asset_keys_by_output_name.items():
+            # This could be sped up by batching
+            output_def = _op_output_def_for_asset_key(self.node_def, asset_key, output_name)
+            result.append(
+                SourceAsset(
+                    key=asset_key,
+                    metadata=output_def.metadata,
+                    io_manager_key=output_def.io_manager_key,
+                    description=output_def.description,
+                )
+            )
+
+        return result
+
+
+def _op_output_def_for_asset_key(
```

Comment:
Doesn't `node_def.resolve_output_to_origin(output_name)` do this for you?

Reply:
oo great call. changing.